pkg/cmd/recommend/trigger.go

```diff
@@ -0,0 +1,144 @@
+package recommend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/gocrane/kubectl-crane/pkg/cmd/options"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+)
+
+var (
+	triggerRecommendExample = `
+# manually trigger the specified recommendation rule
+%[1]s recommend trigger --name workloads-ntzns -n default
+
+# pre-commit
+%[1]s recommend trigger --name workloads-ntzns -n default --dry-run
+`
+)
+
+type RecommendTriggerOptions struct {
+	CommonOptions *options.CommonOptions
+
+	DryRun bool
+	Name   string
+}
+
+func NewRecommendTriggerOptions() *RecommendTriggerOptions {
+	return &RecommendTriggerOptions{
+		CommonOptions: options.NewCommonOptions(),
+	}
+}
+
+func NewCmdRecommendTrigger() *cobra.Command {
+	o := NewRecommendTriggerOptions()
+
+	command := &cobra.Command{
+		Use:     "trigger",
+		Short:   "Manually triggering a recommendation",
+		Example: fmt.Sprintf(triggerRecommendExample, "kubectl-crane"),
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				klog.Infof(fmt.Sprintf("\nExample:\n"+triggerRecommendExample, "kubectl-crane"))
+				return err
+			}
+
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	o.AddFlags(command)
+	o.CommonOptions.AddCommonFlag(command)
+
+	return command
+}
+
+func (o *RecommendTriggerOptions) Validate() error {
+	if err := o.CommonOptions.Validate(); err != nil {
+		return err
+	}
+
+	if len(o.Name) == 0 {
+		return errors.New("please specify the recommend name")
+	}
+
+	if len(*o.CommonOptions.ConfigFlags.Namespace) == 0 {
+		return errors.New("please specify the recommend namespace")
+	}
+
+	return nil
+}
+
+func (o *RecommendTriggerOptions) Complete(cmd *cobra.Command, args []string) error {
+	if err := o.CommonOptions.Complete(cmd, args); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *RecommendTriggerOptions) Run() error {
+	recommend, err := o.CommonOptions.CraneClient.AnalysisV1alpha1().Recommendations(*o.CommonOptions.ConfigFlags.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.New("the recommend doesn't exist, please specify a existed recommend name with --name")
+	}
+
+	// get related recommendation rule
+	recommendationRuleName, ok := recommend.Labels[RecommendationRuleNameLabel]
```

Comment:
No need to get recommendationRule's runNumber ,just set annotation to 0 and exist.